### ISSUE TITLE
Is this expected with no-undef-properties and defineModel in <script setup>?

### ISSUE THREAD

**Issue by dsl101** [vuejs/eslint-plugin-vue#2401]
[This section](https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-generate-no-undef-warnings) of the FAQ mentions `no-undef` warnings when using `defineProps` and `definteEmits`. I don't know if this is the same issue, and I am using vue-eslint-parser > 9.0.0.

**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** v8.56.0 (eslint_d v13.1.2)
- **eslint-plugin-vue version:** 9.20.0
- **Vue version:** 3.4.10
- **vue-eslint-parser version:** 9.4.0
- **Node version:** 18.18.2
- **Operating System:** Windows 10

**Please show your full configuration:**

```json5
module.exports = {
  // https://eslint.org/docs/user-guide/configuring#configuration-cascading-and-hierarchy
  // This option interrupts the configuration hierarchy at this file
  // Remove this if you have an higher level ESLint config file (it usually happens into a monorepos)
  root: true,

  parserOptions: {
    ecmaVersion: 2021, // Allows for the parsing of modern ECMAScript features
  },

  env: {
    node: true,
    browser: true,
    'vue/setup-compiler-macros': true
  },

  // Rules order is important, please avoid shuffling them
  extends: [
    'standard',
    // Base ESLint recommended rules
    'eslint:recommended',

    // Uncomment any of the lines below to choose desired strictness,
    // but leave only one uncommented!
    // See https://eslint.vuejs.org/rules/#available-rules
    'plugin:vue/vue3-essential', // Priority A: Essential (Error Prevention)
    'plugin:vue/vue3-strongly-recommended', // Priority B: Strongly Recommended (Improving Readability)
    'plugin:vue/vue3-recommended', // Priority C: Recommended (Minimizing Arbitrary Choices and Cognitive Overhead)

  ],

  plugins: [
    // https://eslint.vuejs.org/user-guide/#why-doesn-t-it-work-on-vue-files
    // required to lint *.vue files
    'vue'
  ],

  globals: {
    ga: 'readonly', // Google Analytics
    cordova: 'readonly',
    __statics: 'readonly',
    __QUASAR_SSR__: 'readonly',
    __QUASAR_SSR_SERVER__: 'readonly',
    __QUASAR_SSR_CLIENT__: 'readonly',
    __QUASAR_SSR_PWA__: 'readonly',
    process: 'readonly',
    Capacitor: 'readonly',
    chrome: 'readonly'
  },

  // add your custom rules here
  rules: {
    
    // allow async-await
    'generator-star-spacing': 'off',
    // allow paren-less arrow functions
    'arrow-parens': 'off',
    'one-var': 'off',
    'no-void': 'off',
    'multiline-ternary': 'off',

    // 'import/first': 'off',
    'import/named': 'error',
    'import/namespace': 'error',
    'import/default': 'error',
    'import/export': 'error',
    // 'import/extensions': 'off',
    // 'import/no-unresolved': 'off',
    // 'import/no-extraneous-dependencies': 'off',
    
    'prefer-promise-reject-errors': 'off',

    // allow debugger during development only
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',

    // Custom
    'no-return-assign': 'off',  // allows const f = v => thing = v
    'no-console': 'off',
    'quote-props': 'off',
    'eol-last': 'off',
    'comma-dangle': [ 'warn', 'only-multiline' ],
    'no-trailing-spaces': 'off',
    'padded-blocks': 'off',
    'no-multi-spaces': [ 'warn', { ignoreEOLComments: true }],
    'array-bracket-spacing': [ 'warn', 'always', { arraysInArrays: false, objectsInArrays: false }],
    'object-curly-spacing': [ 'warn', 'always', { objectsInObjects: false, arraysInObjects: false }],
    'no-unused-vars': [ 'warn', { args: 'after-used' }],
    'indent': 'off',
    'no-multiple-empty-lines': [ 'warn', { maxBOF: 0, max: 2, maxEOF: 0 }],
    // Vue
    'vue/multi-word-component-names': 'off',
    'vue/max-attributes-per-line': [ 'warn', {
      'singleline': 6,
      'multiline': 3
    }],
    'vue/first-attribute-linebreak': 'off',
    'vue/html-indent': [ 'warn', 2, {
      'attribute': 1,
      'closeBracket': 0,
      'alignAttributesVertically': false,
      'ignores': []
    }],
    'vue/mustache-interpolation-spacing': 'off',
    'vue/singleline-html-element-content-newline': 'off',
    'vue/multiline-html-element-content-newline': [ 'warn', {
        'ignoreWhenEmpty': true,
        'allowEmptyLines': true
    }],
    // Require console to be 'guarded' by conditionals, e.g. conf.logging && console.log(...)
    // 'no-restricted-syntax': [ 'warn', ':not(LogicalExpression) > CallExpression[callee.object.name="console"]' ],
    'dot-notation': 'warn',
    'vue/require-explicit-emits': 'off',
    'comma-spacing': 'warn',
    'space-in-parens': 'warn',
    'new-cap': 'off',
    'vue/attribute-hyphenation': 'off',
    'vue/v-on-event-hyphenation': 'off',
    'vue/no-unused-vars': 'warn',
    'vue/no-undef-properties': 'warn',
  }
}

```

**What did you do?**	

```vue
<template>
  <div>
    {{ model }}
  </div>
</template>

<script setup>
  const model = defineModel({ type: Object })
  console.log('model.value:', model.value)
</script>
```

**What actually happened?**

```
PS D:\...> eslint_d .\src\Test.vue

D:\...\src\Test.vue
  9:36  warning  'value' is not defined  vue/no-undef-properties

✖ 1 problem (0 errors, 1 warning)
```
Showing in editor:
![image](https://github.com/vuejs/eslint-plugin-vue/assets/900542/fa13a88c-4c66-4f9b-9a4b-110ebd9d81b8)

**Comment by FloEdelmann** [comment]
I can't reproduce it: https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxNjU0KwjAQRq8yZNMWagsuQ/QG4gXc1HYqkTQJyaQgIXc3/VG6mzfzffMi682AjDNBOFnVEV4fGkAMcl4HgBhhyhEFKf0XozE7inZLivbQz+h7Jy2BRwp2fdQb7Wn/dIEBR6nxtlAZgT4WOdyfb+wJUvWLG4WNMq+yWFvN3KmAvKjhgNVi3lzZwmrmgkLPeGRzwFabU9BZdbLOWHQkl9M5pS8VpVTa

Please try upgrading to the latest eslint-plugin-vue version, there were some fixes regarding this rule in https://github.com/vuejs/eslint-plugin-vue/releases/tag/v9.20.1.

**Comment by dsl101** [comment]
Wow—no idea how I didn't find that previous bug report. Search-fu is failing me... Sorry for the disruption; upgrade fixed it.
